iso/step_add_floppy.go

```diff
@@ -67,4 +67,42 @@ func (s *StepAddFloppy) Run(_ context.Context, state multistep.StateBag) multist
 	return multistep.ActionContinue
 }
 
-func (s *StepAddFloppy) Cleanup(state multistep.StateBag) {}
+func (s *StepAddFloppy) Cleanup(state multistep.StateBag) {
+	_, cancelled := state.GetOk(multistep.StateCancelled)
+	_, halted := state.GetOk(multistep.StateHalted)
+	if !cancelled && !halted {
+		return
+	}
+
+	ui := state.Get("ui").(packer.Ui)
+	
+	stv := state.Get("vm")
+	if stv == nil {
+		return
+	}
+	vm := stv.(*driver.VirtualMachine)
+	
+	std := state.Get("driver")
+	if std == nil {
+		return
+	}
+	d := std.(*driver.Driver)
+
+
+	ui.Say("Remove floppy image ...")
+
+	ds, err := d.FindDatastore(s.Datastore, s.Host)
+	if err != nil {
+		state.Put("error", err)
+	}
+	vmDir, err := vm.GetDir()
+	if err != nil {
+		state.Put("error", err)
+	}
+
+	deletePath := fmt.Sprintf("%v/packer-tmp-created-floppy.flp", vmDir)
```

Comment:
Can we get the filename from the state instead of hardcoding it?
https://github.com/jetbrains-infra/packer-builder-vsphere/blob/master/iso/step_remove_floppy.go#L33

Reply:
No. Because we could be faced with a situation: when the file not upload, the state will not have information about uploadPath.
https://github.com/jetbrains-infra/packer-builder-vsphere/blob/a00a2ade14e1f0f5c2ce16b9ea1dd85227f8e8e6/iso/step_add_floppy.go#L42-L47